### PR TITLE
fix: inconsistent component order on home screen

### PR DESCRIPTION
**Pull request by tido64**
Implements https://github.com/microsoft/continuity/issues/61#issuecomment-625212973.

### CODE REVIEW

File: android/app/src/main/java/com/sample/manifest/Manifest.kt [hunk 1]
```diff
@@ -2,14 +2,20 @@ package com.sample.manifest
 
 import com.squareup.moshi.JsonClass
 
-typealias Components = Map<String, Component>
+typealias Components = Array<Component>
```
Comment (ArazAbishov) [inline android/app/src/main/java/com/sample/manifest/Manifest.kt:5]: The benefits of having a type alias are not that significant with removal of the `Map`. Thus, I recommend to remove it. Also, it is not common to use `Array` as it doesn't implement an immutable collection API. I would replace with `List<T>` instead.

File: android/app/src/main/java/com/sample/manifest/Manifest.kt [hunk 2]
```diff
@@ -2,14 +2,20 @@ package com.sample.manifest
...
 
 @JsonClass(generateAdapter = true)
 data class Manifest(
     val name: String,
     val displayName: String,
+
+    @Suppress("ArrayInDataClass")
```
Comment (ArazAbishov) [inline android/app/src/main/java/com/sample/manifest/Manifest.kt:12]: I suspect that this warning will be gone if `Array` is replaced with `List<T>` as suggested in the comment above.

File: android/app/src/main/java/com/sample/manifest/Manifest.kt [hunk 3]
```diff
@@ -2,14 +2,20 @@ package com.sample.manifest
...
     val components: Components
 )
 
 @JsonClass(generateAdapter = true)
-data class Component(val displayName: String?)
+data class Component(
+    val appKey: String,
+    val displayName: String?,
+    val initialProperties: Map<String, String>?
```
Comment (ArazAbishov) [inline android/app/src/main/java/com/sample/manifest/Manifest.kt:20]: Can the prop value be `null`? If yes, the type of `initialProperties` needs to reflect that: `Map<String, String?>?`

File: ios/ReactTestApp/ContentView.swift [hunk 4]
```diff
@@ -49,8 +49,8 @@ final class ContentViewController: UITableViewController {
 
         DispatchQueue.global(qos: .userInitiated).async { [weak self] in
             self?.reactInstance.initReact { components in
-                let items: [NavigationLink] = components.map { (name, component) in
-                    (component.displayName ?? name, { self?.navigate(to: component, name: name) })
+                let items: [NavigationLink] = components.map { (component) in
+                    (component.displayName ?? component.appKey, { self?.navigate(to: component) })
```
Comment (ArazAbishov) [inline ios/ReactTestApp/ContentView.swift:53]: I am not sure if I understand this expression correctly, so feel free to dismiss this comment if I am wrong. Shouldn't the `component.appKey` null-checked first and then `displayName`?

File: ios/ReactTestApp/ContentView.swift [hunk 5]
```diff
@@ -132,21 +132,21 @@ final class ContentViewController: UITableViewController {
...
-    private func navigate(to component: Component, name: String) {
+    private func navigate(to component: Component) {
         guard let bridge = reactInstance.bridge,
               let navigationController = navigationController else {
             return
         }
 
         let viewController: UIViewController = {
-            if let viewController = RTAViewControllerFromString(name, bridge) {
+            if let viewController = RTAViewControllerFromString(component.appKey, bridge) {
```
Comment (ArazAbishov) [inline ios/ReactTestApp/ContentView.swift:142]: Shouldn't we fallback to `displayName` here too if we want to keep backwards compat?

File: ios/ReactTestApp/ContentView.swift (hunk 4 above)
Comment (tido64) [reply ios/ReactTestApp/ContentView.swift:53]: `appKey` is what used to be the keys under `components`. It is required.

File: ios/ReactTestApp/ContentView.swift (hunk 5 above)
Comment (tido64) [reply ios/ReactTestApp/ContentView.swift:142]: No, this function tries to retrieve the native view controller given its name.

**Pull request merged**